studioGhibliApi.js

```diff
@@ -0,0 +1,103 @@
+let container;
+let select;
+let opt;
+let ghibliArray;
+
+
+document.addEventListener('DOMContentLoaded', () => {
+    getMovieTitle()
+    let form = document.querySelector('form')
+    form.addEventListener('submit', (event) => {
+        submitReview(event);
+    })
+})
+const getApiData = async () => {
+    const movies = await axios.get('https://ghibliapi.herokuapp.com/films')
+    console.log(movies.data);
+    return movies.data
+}
+
+const submitReview = (event) => {
+    event.preventDefault()
+    opt = getSelectedMovie(select).text.bold()
+    let list = document.querySelector('ul')
+    let noInput = document.querySelector('#error')
+    let textInput = document.querySelector('#textField').value
+    let newReview = document.createElement('li')
+    if (textInput === '') {
+        noInput.innerText = `Please enter a valid review`
+        newReview.innerText = ''
+    } else {
+        noInput.innerText = ''
+        newReview.innerHTML = ` ${opt}: ${textInput}`
+        list.appendChild(newReview)
+    }
+    document.querySelector('#textField').value = ''
+}
+
+const pullFromHtml = async () => {
+    return select = document.querySelector('select')
+}
+
+const getMovieTitle = async () => {
+    select = await pullFromHtml();
+    ghibliArray = await getApiData();
+    for (let i = 0; i < ghibliArray.length; i++) {
+        let options = document.createElement('option')
+        options.value = ghibliArray[i].title;
+        options.innerText = ghibliArray[i].title;
+        select.appendChild(options)
+    }
+    getMovieInfo(ghibliArray)
+}
+const getSelectedMovie = (select) => {
+    let option;
+    for (let i = 0; i < select.length; i++) {
+        option = select[i]
+        if (option.selected === true) {
+            break;
+        }
+    }
+    return option
+}
+
+const getMovieInfo = async (param) => {
+    select = await pullFromHtml();
+    select.addEventListener('change', () => {
+        replaceSelection()
+        opt = getSelectedMovie(select);
```

Comment:
There is a better option for handling this. You are storing the film `title` (alternatively you could store its `id`) in the `optionElement.value`. When the user picks an option the `selectElement.value` will change to have the film's `title` (or `id`). [Check this](https://www.w3schools.com/jsref/prop_select_value.asp) click `Try it yourself`. This would allow you identify the chosen movie without having to loop over the options, which `getSelectedMovie()` is doing. Currently your listener for `change` on the select has a runtime complexity of `O(n)` where `n` is the total number of options. This function could be `O(1)` constant time. One part is what I mentioned above, another has to do with a hashmap or an object. Give it a try. After that talk to @MikePursuit and compare your approaches.

Besides the better approach that I mention above, our intent here was for you to make a network request to `/films/<film-id>` after grabbing the id of the selected movie to get the details of the movie. Doing this would bump your grade for standard `EF.10.c: Make requests to APIs` up.